build/yamls/flow-visibility/base/clickhouse.yml

```diff
@@ -36,7 +36,100 @@ spec:
               volumeMounts:
                 - name: clickhouse-configmap-volume
                   mountPath: /docker-entrypoint-initdb.d
+                - name: clickhouse-storage-volume
+                  mountPath: /var/lib/clickhouse
           volumes:
             - name: clickhouse-configmap-volume
               configMap:
                 name: $(CLICKHOUSE_CONFIG_MAP_NAME)
+            - name: clickhouse-storage-volume
+              emptyDir:
+                medium: Memory
+                sizeLimit: 8Gi
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    app: flow-visibility
+  name: clickhouse-monitor
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  labels:
+    app: flow-visibility
+  name: clickhouse-monitor-role
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - pods
+    verbs:
+      - list
+  - apiGroups:
+      - ""
+    resources:
+      - pods/log
+    verbs:
+      - get
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  labels:
+    app: flow-visibility
+  name: clickhouse-monitor-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: clickhouse-monitor-role
+subjects:
+  - kind: ServiceAccount
+    name: clickhouse-monitor
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  labels:
+    app: clickhouse-monitor
+  name: clickhouse-monitor
+spec:
+  schedule: "* * * * *"
+  successfulJobsHistoryLimit: 1
+  failedJobsHistoryLimit: 1
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels:
+            app: clickhouse-monitor
+        spec:
+          serviceAccountName: clickhouse-monitor
+          containers:
+          - name: clickhouse-monitor
+            image: flow-visibility-clickhouse-monitor
+            env:
+              - name: CH_USERNAME
+                valueFrom:
+                  secretKeyRef: 
+                    name: clickhouse-secret
+                    key: username
+              - name: CH_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    name: clickhouse-secret
+                    key: password
+              - name: SVC_HOST
+                value: "clickhouse-clickhouse.flow-visibility.svc.cluster.local"
+              - name: SVC_PORT
+                value: "9000"
+              - name: TABLE_NAME
+                value: "default.flows"
+              - name: MV_NAMES
+                value: "default.flows_pod_view default.flows_node_view default.flows_policy_view"
+              - name: NAMESPACE
+                value: "flow-visibility"
```

Comment:
I feel like you could use the downward Pod API here for the default value

Reply:
Updated, thanks